### ISSUE TITLE
cmd/gc: inline func expanding seems to be broken

### ISSUE THREAD

**Issue by mattn** [golang/go#10066]
Maybe typecheck on cmd/internal/gc is broken?

### `/path/to/gopath/github.com/mattn/foobar/foobar.go`
```go
package foobar

import (
	"log"
)

func Do() {
	Do2()
}

func Do2() {
	println(log.Ldate | log.Ltime | log.Lshortfile)
}
```
### `/path/to/gopath/github.com/mattn/foobar/cmd/foobar/main.go`
```go
package main

import (
	"github.com/mattn/foobar"
)

func main() {
	foobar.Do()
}
```
### build result
`go build main.go`
```
.\main.go:4: undefined: log.Ltime
.\main.go:4: undefined: log.Lshortfile
.\main.go:8: tempname called with nil type
```

**Comment by josharian** [comment]
Eep. Can you narrow it down to a particular commit?

**Comment by mattn** [comment]
Yes, just doing that. :)

**Comment by mattn** [comment]
Hmm, maybe my environment is wrong. Sorry noisy. When I figure out, I'll reopen this.

**Comment by mattn** [comment]
Ooops, I was checked this on linux, But today, I could reproduce this on windows too. So re-open. Anyone can repro?

**Comment by mattn** [comment]
Below is a minimum code.

### `/path/to/gopath/github.com/mattn/foobar/foobar.go`
```go
package foobar

import (
	"log"
)

func Do() {
	Do2()
}

func Do2() {
	println(log.Ldate | log.Ltime | log.Lshortfile)
}
```
### `/path/to/gopath/github.com/mattn/foobar/cmd/foobar/main.go`
```go
package main

import (
	"github.com/mattn/goemon"
)

func main() {
	goemon.Do()
}
```
### build result
`go build main.go`
```
.\main.go:4: undefined: log.Ltime
.\main.go:4: undefined: log.Lshortfile
.\main.go:8: tempname called with nil type
```

**Comment by mattn** [comment]
Maybe, inline optimization is broken.

**Comment by mattn** [comment]
I added workaround for this issue into goemon, so I changed issue description.

https://github.com/mattn/goemon/commit/d4f81d61b5f0510c68efe86b9738e62911ba836f

**Comment by minux** [comment]
This is a side effect of fixing #9076. It has nothing to do with transitive inlining.

**Comment by mattn** [comment]
I'm thinking inline must look code origin is same as: https://github.com/golang/go/blob/master/src/cmd/internal/gc/inl.go#L164

**Comment by minux** [comment]
The problem is the export format changes after the fix for #9076.

It was:
```
$$
package issue10066
        import log "log"
        import runtime "runtime"
        func @"".Do ()
        func @"".Do2 () { println(0x13) }
        func @"".init ()
        const @"log".Ldate = 0x1

$$
```

Now it's:
```
$$
package issue10066
        import runtime "runtime"
        import log "log"
        func @"".Do () { @"".Do2() }
        func @"".Do2 () { println(@"log".Ldate | @"log".Ltime | @"log".Lshortfile) }
        func @"".init ()
        const @"log".Ldate = 0x1

$$
```
